modules/context/repo.go

```diff
@@ -820,3 +832,56 @@ func UnitTypes() macaron.Handler {
 		ctx.Data["UnitTypeExternalTracker"] = models.UnitTypeExternalTracker
 	}
 }
+
+// IssueTemplatesFromDefaultBranch checks for issue templates in the repo's default branch
+func (ctx *Context) IssueTemplatesFromDefaultBranch() []api.IssueTemplate {
+	var issueTemplates []api.IssueTemplate
+	if ctx.Repo.Commit == nil {
+		var err error
+		ctx.Repo.Commit, err = ctx.Repo.GitRepo.GetBranchCommit(ctx.Repo.Repository.DefaultBranch)
+		if err != nil {
+			return issueTemplates
```

Comment:
why not return error message ?

Reply:
I think that would then cause an error if the repo is bare.